bus-mapping/src/circuit_input_builder.rs

```diff
@@ -181,101 +217,136 @@ pub enum CallType {
     Create2,
 }
 
-impl CallType {
+impl CallKind {
     fn is_create(&self) -> bool {
         matches!(self, Self::Create | Self::Create2)
     }
 }
 
-impl TryFrom<OpcodeId> for CallType {
+impl TryFrom<OpcodeId> for CallKind {
     type Error = Error;
 
     fn try_from(op: OpcodeId) -> Result<Self, Self::Error> {
         Ok(match op {
-            OpcodeId::CALL => CallType::Call,
-            OpcodeId::CALLCODE => CallType::CallCode,
-            OpcodeId::DELEGATECALL => CallType::DelegateCall,
-            OpcodeId::STATICCALL => CallType::StaticCall,
-            OpcodeId::CREATE => CallType::Create,
-            OpcodeId::CREATE2 => CallType::Create2,
+            OpcodeId::CALL => CallKind::Call,
+            OpcodeId::CALLCODE => CallKind::CallCode,
+            OpcodeId::DELEGATECALL => CallKind::DelegateCall,
+            OpcodeId::STATICCALL => CallKind::StaticCall,
+            OpcodeId::CREATE => CallKind::Create,
+            OpcodeId::CREATE2 => CallKind::Create2,
             _ => return Err(Error::OpcodeIdNotCallType),
         })
     }
 }
 
-/// Context of a Call during a [`Transaction`] which can mutate in an
-/// [`ExecStep`].
+/// Circuit Input related to an Ethereum Call
 #[derive(Debug)]
-pub struct CallContext {
+pub struct Call {
     /// Type of call
-    call: CallType,
+    kind: CallKind,
     /// This call is being executed without write access (STATIC)
     is_static: bool,
```

Comment:
This can already be extracted from `CallKind` no? If we have a `CallKind::StaticCall`. Are we duplicating info?

Reply:
Here I'd like to mention two things
1. The `is_static` flag is not being used right now, but I think it may be useful for the circuit.
2. Notice that in the EVM, once you are in a `STATICCALL`, any further *CALL* will also be static!  For example, you can make a `CALL` inside a `STATICCALL`, and that's valid (as long as the CALL doesn't transfer value and doesn't write to the state).  So this flag can be true while the `CallKind` is different than `StaticCall`.

Reply:
But then the only possible Call Kinds inside of the StaticCall would also be Static too no?

If so, this would still apply. Anyway, it's fine if you want to keep it. I just suggested it so that there was less things to manage and worry about in the struct.

Reply:
> But then the only possible Call Kinds inside of the StaticCall would also be Static too no?

From what I read in the spec, you can do a `CALL` with value = 0 within a `STATICCALL`: https://eips.ethereum.org/EIPS/eip-214
So you can have `{kind: Call, is_static: true`}

Reply:
True. As discussed in the call, my bad.